src/web/client/common/worker/webworker.js

```diff
@@ -0,0 +1,165 @@
+/*
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License. See License.txt in the project root for license information.
+ */
+
+// eslint-disable-next-line no-undef
+self.window = self;
+// eslint-disable-next-line @typescript-eslint/no-var-requires
+const fluid = require("fluid-framework");
+// eslint-disable-next-line @typescript-eslint/no-var-requires
+const { AzureClient } = require("@fluidframework/azure-client");
+// eslint-disable-next-line @typescript-eslint/no-var-requires
+const DataverseTokenProvider = require("../DataverseTokenProvider");
+
+const { SharedMap, ConnectionState } = fluid;
+
+const objectTypes = [
+    SharedMap
+]
+
+const containerSchema = {
+    dynamicObjectTypes: objectTypes,
+    initialObjects: {
+        sharedState: SharedMap,
+    },
+};
+
+let initial = false;
+
+class AzureFluidClient {
+    static _clientInstance;
+    static _container;
+    static _audience;
+    static _userSharedMap;
+
+    static getInstance(config) {
+        if (!this._clientInstance) {
+            const afrClientProps = {
+                connection: {
+                    type: "remote",
+                    tenantId: config.swptenantId,
+                    tokenProvider: new DataverseTokenProvider(
+                        config.swpAccessToken,
+                        () => this.fetchAccessToken()
+                    ),
+                    endpoint: config.discoveryendpoint,
+                },
+            };
+
+            AzureFluidClient._clientInstance = new AzureClient(afrClientProps);
+        }
+        return this._clientInstance;
+    }
+
+    static async fetchContainerAndService(config, id) {
+        if (
+            this._container?.connectionState !== ConnectionState.Connected
+        ) {
+            const azureClient = this.getInstance(config);
+            const { container, services } = await azureClient.getContainer(
+                id,
+                containerSchema
+            );
+            if (container.connectionState !== ConnectionState.Connected) {
+                await new Promise((resolve) => {
+                    container.once("connected", () => {
+                        resolve();
+                    });
+                });
+            }
+            this._container = container;
+            this._audience = services.audience;
+            this._userSharedMap = container.initialObjects.sharedState;
+        }
+        return {
+            container: this._container,
+            audience: this._audience,
+            map: this._userSharedMap,
+        };
+    }
+}
+
+async function loadContainer(config, id, swpId) {
+    try {
+        const { container, audience, map } =
+            await AzureFluidClient.fetchContainerAndService(config, swpId);
+
+        const existingMembers = audience.getMembers();
+
+        // TODO: insert user with entity id in container
```

Comment:
what is this about?

Reply:
rn I'm not adding user entityId joined from vscode side! we need to have a discussion what will be the formate in which we will add it